### ISSUE TITLE
[PSyData] Support for LFRic/dynamo

### ISSUE THREAD

**Issue by hiker** [stfc/PSyclone#646]
ATM dynamo3/LFRic is only partially creating correct code for LFRic extraction:
```
      CALL psy_data%PreStart("unknown-module", "setval_c", 1, 3)
      CALL psy_data%PreDeclareVariable("f2", f2)
      CALL psy_data%PreDeclareVariable("cell_post", cell)
      CALL psy_data%PreDeclareVariable("df_post", df)
      CALL psy_data%PreDeclareVariable("f3_post", f3)
      CALL psy_data%PreEndDeclaration
      CALL psy_data%ProvideVariable("f2", f2)
      CALL psy_data%PreEnd
      DO df=1,undf_any_space_1_f5
        f5_proxy%data(df) = 0.0
      END DO
      DO df=1,undf_any_space_1_f2
        f2_proxy%data(df) = 0.0
      END DO
      DO cell=1,f3_proxy%vspace%get_ncell()
        !
        CALL testkern_code_w2_only(nlayers, f3_proxy%data, """ + \
        """f2_proxy%data, ndf_w2, undf_w2, map_w2(:,cell))
      END DO
      CALL psy_data%PostStart
      CALL psy_data%ProvideVariable("cell_post", cell)
      CALL psy_data%ProvideVariable("df_post", df)
      CALL psy_data%ProvideVariable("f3_post", f3)
      CALL psy_data%PostEnd
```
Parameters like `nlayer`, `ndf_w2`, `undf_w2`, `map_w2` are not saved (this test also shows #637, e.g. f5 is not stored either)

**Comment by TeranIvy** [comment]
Hi @hiker, the reason for this are many LFRic API-specific rules that go with arguments specified via metadata. I am currently capturing these rules as part of PSyke work - see e.g. issue #234 - so we will need to coordinate.

**Comment by TeranIvy** [comment]
Hi @hiker, the reason for this are many LFRic API-specific rules that go with arguments specified via metadata so this is really an "enhancement" rather than a "bug". I am currently capturing these rules as part of PSyke work - see e.g. issue #234 - so we will need to coordinate.

**Comment by hiker** [comment]
I only opened this ticket to document this issue, I won't touch it and leave it to the experts :)
That LFRic now creates some code is a bit of a coincidence - just the base class doing something. The LFRic specific-transformation (or node) needs to be properly implemented.

**Comment by TeranIvy** [comment]
Hi @hiker, I am just looking at your PR #650 and the implemented functionality looks great :-) I will take this on as soon as your PR is merged. Many thanks for all your hard work!

**Comment by hiker** [comment]
This has been fixed, outstanding is support for builtins (#637).
